source/fab/steps/preprocess.py

```diff
@@ -126,6 +126,7 @@ def fortran_preprocessor(preprocessor=None, source=None,
     Params as per :class:`~fab.steps.preprocess.PreProcessor`.
 
     """
+    # todo: we want to add -P ... IF it's not already there
```

Comment:
This makes assumptions about what arguments the preprocessor takes. Ultimately there needs to be way of representing each tool implementation with a standardised set of data describing the correct arguments to pass for various functions.

Reply:
Yes, that definitely needs addressing. Perhaps a chat sometime soon?